src/test/RASTER_IMAGE_DATA_PERFORMANCE.test.ts

```diff
@@ -0,0 +1,383 @@
+                let payload = CARTA.RegisterViewer.encode(message).finish();
+                let eventData = new Uint8Array(32 + 4 + payload.byteLength);
+
+                eventData.set(Utility.stringToUint8Array("REGISTER_VIEWER", 32));
+                eventData.set(new Uint8Array(new Uint32Array([1]).buffer), 32);
+                eventData.set(payload, 36);
+
+                Connection.send(eventData);
+            } else {
+                console.log(`Can not open a connection.`);
+            }
+            done();
+        };
+    }, connectionTimeout);
+
+    test(`connect to CARTA "${testServerUrl}" & ...`, 
+    done => {
+        // While receive a message in the form of arraybuffer
+        Connection.onmessage = (event: MessageEvent) => {
+            const eventName = Utility.getEventName(new Uint8Array(event.data, 0, 32));
+            if (eventName === "REGISTER_VIEWER_ACK") {
+                expect(event.data.byteLength).toBeGreaterThan(0);
+                const eventData = new Uint8Array(event.data, 36);
+                expect(CARTA.RegisterViewerAck.decode(eventData).success).toBe(true);
+                
+                done();
+            }
+        };
+    }, connectionTimeout);
+
+    describe(`access directory`, () => {
+        [[expectRootPath], [testSubdirectoryName]
+        ].map(
+            ([dir]) => {
+                test(`assert the directory "${dir}" opens.`, 
+                done => {
+                    // Preapare the message
+                    let message = CARTA.FileListRequest.create({directory: testSubdirectoryName});
+                    let payload = CARTA.FileListRequest.encode(message).finish();
+                    let eventDataTx = new Uint8Array(32 + 4 + payload.byteLength);
+            
+                    eventDataTx.set(Utility.stringToUint8Array("FILE_LIST_REQUEST", 32));
+                    eventDataTx.set(new Uint8Array(new Uint32Array([1]).buffer), 32);
+                    eventDataTx.set(payload, 36);
+            
+                    Connection.send(eventDataTx);
+            
+                    // While receive a message
+                    Connection.onmessage = (event: MessageEvent) => {
+                        let eventName = Utility.getEventName(new Uint8Array(event.data, 0, 32));
+                        if (eventName === "FILE_LIST_RESPONSE") {
+                            expect(event.data.byteLength).toBeGreaterThan(0);
+                            let eventData = new Uint8Array(event.data, 36);
+                            expect(CARTA.FileListResponse.decode(eventData).success).toBe(true);
+            
+                            done();
+                        }
+                    };
+                }, openFileTimeout);
+            }
+        );
+    });
+
+    describe(`prepare the files`, () => {
+        [ 
+         ["cluster_00128.fits", {xMin: 0, xMax:   128, yMin: 0, yMax:   128},  1, CARTA.CompressionType.ZFP, 18, 4],
```

Comment:
it doesn't look like you're using the last three components of this in the test itself (they're hard-coded in the request below)

Reply:
The performance tests should also test how performance scales with subsets, so perhaps some tests should have 1, 2, 3 or 4 subsets.

Reply:
OK, sure. I will try to make test case for this.